File: ShadowFrameworkJS/src/shadow/geometry/data/SFFixedFloatUnit8.js

var SFFixedFloatUnit8_BIT_SIZE=8;
var SFFixedFloatUnit8_MULTIPLIER=0.004;
var SFFixedFloatUnit8_BACKMULTIPLIER=250;

function SFFixedFloatUnit8(f){
	this.multiplier=SFFixedFloatUnit8_MULTIPLIER;
	this.backmultiplier=SFFixedFloatUnit8_BACKMULTIPLIER;
	if(!(f==undefined)){
		this.setFloat(f);
	}
}

inherit(SFFixedFloatUnit8,SFGenericFixedFloat);


SFFixedFloatUnit8.prototype["getFloat"]=function(){
			return this.value*this.multiplier;
		};

SFFixedFloatUnit8.prototype["clone"]=function(){
	return new SFFixedFloatUnit8();
};

SFFixedFloatUnit8.prototype["getBitSize"]=function(){
	return SFFixedFloatUnit8_BIT_SIZE;
};

File: trunk/sfoglintegration/SF20Data/src/shadow/system/data/formats/SFFixedFloatUnit8.java
package shadow.system.data.formats;

import shadow.system.data.objects.SFGenericFixedFloat;


public class SFFixedFloatUnit8 extends SFGenericFixedFloat{

	private static int BIT_SIZE=8;
	private static float MULTIPLIER=0.008f;
	private static float BACKMULTIPLIER=125f;
	
	public SFFixedFloatUnit8() {
		super(MULTIPLIER,BACKMULTIPLIER);
	}
	
	public SFFixedFloatUnit8(float multiplier, float backmultiplier) {
		super(multiplier,backmultiplier);
	}

	public SFFixedFloatUnit8(float f) {
		super(MULTIPLIER,BACKMULTIPLIER);
		setFloat(f);
	}
	
	@Override
	public SFFixedFloatUnit8 copyDataObject() {
		return new SFFixedFloatUnit8();
	}
	
	@Override
	public int getBitSize() {
		return BIT_SIZE;
	}
}
